UnitedIncome/components
# McGonagall Docs Re-Org / MVP Generator

### Description
> Please provide a brief description of your changes below.

Improved the McGonagall Documentation by re-organizing it and providing some additional context. Also added a `yarn mcgonagall` command that allows you to generate a new flow within the component library, this allows us to keep the docs consistent in an open source environment even though our internal generator will exist within the client repo.

I've also fixed the example for the xState visualizer so you can just copy and paste it straight in to see it working. 

### Testing Instructions
> What should be done in order to test this pull request, or re-create its effects.

* Run the `yarn mcgonagall` command and create a flow.
* Read the docs by going to `Templates/McGonagall` and click `Docs`.
* Paste the generated flow into the client and fix the imports.
* You should have a working flow and the docs should be enough to get you up and running.

### Comments
> Any additional comments that will aid with the review process.

I'll move the same generator into client if this looks good to everyone. There's less todo items in this version so it's quite speedy to get it up and running.

Give this a read and let me know if you have a good understanding as to what's going on with McGonagall. The idea is to let new people read this documentation so they can start working with it right away.

- [JIRA Ticket](https://unitedincome.atlassian.net/browse/app-*)
